grade-management/Ahk.GradeManagement.Data/IAssignmentRepository.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Ahk.GradeManagement.Data.Entities;
+
+namespace Ahk.GradeManagement.Data
+{
+    public interface IAssignmentRepository
```

Comment:
A repository mintáról beszélgethetünk, de nem teljesen érzem a létjogosultságát. Ugyan mi most pont azt csináljuk, hogy lecseréljük a cosmosdb-t mssql-re, de közben egy csomó üzleti funkciót hozzáírunk. Tisztán DAL réteg csere ritkán szokott lenni az alkalmazás fejlesztések során. 

Mivel az EF már lényegében a repository mintát valósítja meg (DbContect a Unit-of-Work, a DbSet pedig a Repository), nem szoktam az EF fölé plusz repository réteget húzni, mert felesleges kötöttségekkel jár, amit egyébként nem fog tudni a jövőben előnyre fordítani.

Ha  egyébként a Repositorykat Service-nek hívnánk és ezt tekintenénk a BLL rétegnek máris jobban festene a kép. (Ezt egyébként Ákos az adatvez tananyagban is összemosta).

Szóval a javaslatom, hogy a repository-k legyenek service-ek és tekintsük ezeket a BLL-nek.